### PR TITLE
Add custom user model

### PR DESCRIPTION
**Pull request by MargitStar**

### CODE REVIEW

File: music_shop/genre/views.py [hunk 1]
```diff
@@ -14,6 +15,20 @@ def list(self, request, *args, **kwargs):
...
+        genre = get_object_or_404(queryset, pk=pk)
+        serializer = GenreSerializer(genre, context={"request": request})
         return Response(serializer.data, status=status.HTTP_200_OK)
+
+    @action(
+        detail=True, methods=["get"], permission_classes=(permissions.IsAuthenticated,)
+    )
+    def like(self, request, pk=None):
+        queryset = Genre.objects.all()
+        genre = get_object_or_404(queryset, pk=pk)
```
Comment (vadim-denisov) [inline music_shop/genre/views.py:27]: Does it work? 
```suggestion
        genre = get_object_or_404(Genre, pk=pk)
```

File: music_shop/user/managers.py [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+        return user
+
+    def create_staffuser(self, username, password=None):
+        if not username:
+            raise ValueError("User must have an username")
+        if not password:
+            raise ValueError("User must have a password")
+
+        user = self.model(username=username)
+        user.set_password(password)
```
Comment (vadim-denisov) [inline music_shop/user/managers.py:38]:
```suggestion
```

File: music_shop/user/managers.py [hunk 3]
```diff
@@ -0,0 +1,43 @@
+from django.contrib.auth.base_user import BaseUserManager
+
+
+class CustomUserManager(BaseUserManager):
+    def create_user(self, username, password=None):
+        if not username:
+            raise ValueError("User must have an username")
+        if not password:
+            raise ValueError("User must have a password")
```
Comment (vadim-denisov) [inline music_shop/user/managers.py:9]: Can be move to private method / function.

File: music_shop/user/migrations/0003_customuser_favourite_author.py [hunk 4]
```diff
@@ -0,0 +1,19 @@
+# Generated by Django 3.2.6 on 2021-09-13 12:12
```
Comment (vadim-denisov) [inline music_shop/user/migrations/0003_customuser_favourite_author.py:1]: Squash migrations

File: music_shop/user/models.py [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+
+    objects = CustomUserManager()
+
+    @staticmethod
+    def has_perm(perm, obj=None):
+        return True
+
+    @staticmethod
+    def has_module_perms(app_label):
+        return True
```
Comment (vadim-denisov) [inline music_shop/user/models.py:33]: Why should we define this methods?

Comment (MargitStar) [reply music_shop/user/models.py:33]: `has_perm` checks whether the user has a specific permission
`has_module_perms` checks whether the user has any permissions for the app

**Pull request merged**